modules/ServerAPI/src/server/routes/FileServices.coffee

```diff
@@ -200,15 +459,134 @@ setupRoutes = (app, loginRoutes, requireLogin) ->
 exports.setupAPIRoutes = (app, loginRoutes) ->
 	setupRoutes app, loginRoutes, false
 	app.delete '/dataFiles/*', exports.deleteFile
+	app.post '/api/moveDataFiles', exports.moveDataFiles
 
 exports.setupRoutes = (app, loginRoutes) ->
 	setupRoutes app, loginRoutes, true
 	app.delete '/dataFiles/*', loginRoutes.ensureAuthenticated, exports.deleteFile
+	app.post '/api/moveDataFiles',  loginRoutes.ensureAuthenticated, exports.moveDataFiles
 
 exports.deleteFile = (req, resp) ->
 	# Not implemented currently. We added this route in order to fully support the jquery file upload plugin
 	# which requires a delete route in order to function correctly with maxNumberOfFiles.
 	# See details here: https://github.com/mcneilco/acas/pull/1085
 	console.log 'Got DELETE request for data file: ' + req.params[0] + ' from user: ' + req.user.username
 	console.log 'Not deleting file, just returning 200 OK'
-	resp.send 200
\ No newline at end of file
+	resp.send 200
+
+
+exports.moveDataFiles = (req, resp) ->
+	deleteSourceFileOnSuccess = if req.query?.deleteSourceFileOnSuccess? then req.query.deleteSourceFileOnSuccess == "true" else true
+	files = await exports.moveDataFilesInternal(req.body, deleteSourceFileOnSuccess)
+	resp.json files
+
+exports.moveDataFilesInternal = (files, deleteSourceFileOnSuccess) ->
+	return await exports.fileHandler.moveFiles(files, deleteSourceFileOnSuccess)
+
+
+exports.migrateCmpdRegBulkLoaderFilesToSubfolders = () ->
+	console.log "About to migrate cmpdReg bulk loader files to subfolders"
+	cmpdRegBulkLoaderRoutes = require('./CmpdRegBulkLoaderRoutes.js')
+	bulkLoadFiles = await cmpdRegBulkLoaderRoutes.getBulkloadFilesInternal()
+	console.log "There are #{bulkLoadFiles.length} bulk load files in the DB"
+	bulkLoadSubFolderFiles = await exports.localFileHandler.listFiles(cmpdRegBulkLoaderRoutes.BULKLOAD_SUB_FOLDER, false)
+	console.log "There are #{bulkLoadSubFolderFiles.length} bulk load files in the local bulkload subfolder"
+	if exports.fileHandler instanceof ExternalFileHandler
+		externalBulkLoadSubFolderFiles = bulkLoadSubFolderFiles.concat(await exports.fileHandler.listFiles(cmpdRegBulkLoaderRoutes.BULKLOAD_SUB_FOLDER, false))
```

Comment:
Then we are looking in the external file location and checking for more properly housed files?